### PR TITLE
Add retries if the stratum hostname is not found

### PR DESCRIPTION
**Pull request by SatsForFreedom**
After restoring the Wi-Fi connection due to a power outage, the router may still try to connect to the internet. Therefore, it will fail to solve the IP address, and then the *ipaddr returns a NULL pointer. This will crash the firmware when it reads from a NULL pointer.

### CODE REVIEW

**Comment by benjamin-wilson** [comment]
Why was bDNSFound changed from a bool to an int?

**Comment by SatoshyBoy** [comment]
Because I need 3 states: waiting, success and failure.

File: main/tasks/stratum_task.c [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 static const char *TAG = "stratum_task";
 static ip_addr_t ip_Addr;
-static bool bDNSFound = false;
+static int8_t bDNSFound = 0;
```
Comment (johnny9) [inline main/tasks/stratum_task.c:21]: this variable is no longer a bool, give it a clearer name and use and enum for the different states or have a second bool for tracking dns progress

File: components/bm1397/bm1366.c [hunk 2]
```diff
@@ -704,6 +704,11 @@ task_result *BM1366_proccess_work(void *pvParameters)
 
     GlobalState *GLOBAL_STATE = (GlobalState *)pvParameters;
 
+    if (GLOBAL_STATE->valid_jobs[rx_job_id] == 0) {
```
Comment (johnny9) [inline components/bm1397/bm1366.c:707]: create a new PR for this

**Review by johnny9 — CHANGES_REQUESTED** [review]
Don't create merge commits. When updating your PR rebase instead.

File: components/bm1397/bm1366.c (hunk 2 above)
Comment (johnny9) [reply components/bm1397/bm1366.c:707]: I merged this piece already. Can you update the PR without it?

**Pull request closed without merging**